content/en/docs/concepts/scheduling-eviction/topology-spread-constraints.md

```diff
@@ -129,17 +129,29 @@ your cluster. Those fields are:
   for more details.
 
 - **matchLabelKeys** is a list of pod label keys to select the pods over which
-  spreading will be calculated. The keys are used to lookup values from the pod labels, those key-value labels are ANDed with `labelSelector` to select the group of existing pods over which spreading will be calculated for the incoming pod. Keys that don't exist in the pod labels will be ignored. A null or empty list means only match against the `labelSelector`.
-
-  With `matchLabelKeys`, users don't need to update the `pod.spec` between different revisions. The controller/operator just needs to set different values to the same `label` key for different revisions. The scheduler will assume the values automatically based on `matchLabelKeys`. For example, if users use Deployment, they can use the label keyed with `pod-template-hash`, which is added automatically by the Deployment controller, to distinguish between different revisions in a single Deployment.
+  spreading will be calculated. The keys are used to lookup values from the pod labels,
+  those key-value labels are ANDed with `labelSelector` to select the group of existing
+  pods over which spreading will be calculated for the incoming pod. The same key is
+  forbidden to exist in both MatchLabelKeys and LabelSelector. MatchLabelKeys cannot
+  be set when LabelSelector isn't set. Keys that don't exist in the pod labels will be
+  ignored. A null or empty list means only match against the `labelSelector`.
+
+  With `matchLabelKeys`, users don't need to update the `pod.spec` between different revisions.
+  The controller/operator just needs to set different values to the same `label` key for different
+  revisions. The scheduler will assume the values automatically based on `matchLabelKeys`. For
+  example, if users use Deployment, they can use the label keyed with `pod-template-hash`, which
```

Comment:
Add link to documentation https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#pod-template-hash-label

Reply:
Done